docs/reference/elements/py-config.md

```diff
@@ -0,0 +1,242 @@
+
+- JSON config using the `type` attribute.
+```html
+<py-config type="json">
+  {
+    "autoclose_loader": true,
+    "runtimes": [{
+      "src": "https://cdn.jsdelivr.net/pyodide/v0.21.2/full/pyodide.js",
+      "name": "pyodide-0.21.2",
+      "lang": "python"
+    }]
+  }
+</py-config>
+```
+
+- Use of the `src` attribute:
+```
+<py-config src="./custom.toml"></py-config>
+```
+where `custom.toml` contains
+
+```
+autoclose_loader = true
+[[runtimes]]
+src = "https://cdn.jsdelivr.net/pyodide/v0.21.2/full/pyodide.js"
+name = "pyodide-0.21.2"
+lang = "python"
+```
+
+- JSON using the `type` attribute.
+```
+<py-config type="json" src="./custom.json"></py-config>
+```
+where `custom.json` contains
+
+```
+{
+  "autoclose_loader": true,
+  "runtimes": [{
+    "src": "https://cdn.jsdelivr.net/pyodide/v0.21.2/full/pyodide.js",
+    "name": "pyodide-0.21.2",
+    "lang": "python"
+  }]
+}
+```
+
+One can also use both i.e pass the config from `src` attribute as well as specify it as `inline`. So the following snippet is also valid:
+
+```
+<py-config src="./custom.toml">
+  paths = ["./utils.py"]
+</py-config>
+```
+
+This can also be done via JSON using the `type` attribute.
+
+```
+<py-config type="json" src="./custom.json">
+  {
+    "paths": ["./utils.py"]
+  }
+</py-config>
+```
+
+Note: While the `<py-config>` tag supports both TOML and JSON, one cannot mix the type of config passed from 2 different sources i.e. the case when inline config is in TOML format while config from src is in JSON format is NOT allowed. Similarly for the opposite case.
+
+---
+
+This is helpful in cases where a number of applications share a common configuration (which can be supplied via `src`), but their specific keys need to be customised and overridden.
+
+The keys supplied through `inline` override the values present in config supplied via `src`.
+
+One can also declare dependencies so as to get access to many 3rd party OSS packages that are supported by PyScript.
+You can also link to `.whl` files directly on disk like in our [toga example](https://github.com/pyscript/pyscript/blob/main/examples/toga/freedom.html).
+
+```
+<py-config>
+  packages = ["./static/wheels/travertino-0.1.3-py3-none-any.whl"]
+</py-config>
+```
+
+OR in JSON like
+
+```
+<py-config type="json">
+  {
+    "packages": ["./static/wheels/travertino-0.1.3-py3-none-any.whl"]
+  }
+</py-config>
+```
+
+If your `.whl` is not a pure Python wheel, then open a PR or issue with [pyodide](https://github.com/pyodide/pyodide) to get it added [here](https://github.com/pyodide/pyodide/tree/main/packages).
+
+If there's enough popular demand, the pyodide team will likely work on supporting your package. Regardless, things will likely move faster if you make the PR and consult with the team to get unblocked.
+
+For example, NumPy and Matplotlib are available. Notice here we're using `<py-script output="plot">`
```

Comment:
Very minor comment... For readability, I'd move the "Notice here... " after the example. Maybe a personal thing though...